### ISSUE TITLE
Allow item click and selection event fire in multiselect Grid on SPACE key

### ISSUE THREAD

**Issue by TatuLund** [vaadin/flow-components#8052]
### What is the problem?

When pressing SPACE key in Grid in MultiSelect mode, the item click and selection events are fired only when the selection checkbox cell is focused, but not when the focus in some other cell. That makes Grid sometimes cumbersome to operate.  It is possible to workaround this problem with the following JavaScript call:

```
grid.getElement().executeJs("""
$0.$.scroller.addEventListener('keyup', (e) => {
  if (e.key === " ") { 
    $0._focusedCell.click();
  }
});
""", grid.getElement());
```

### Browsers

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Safari on iOS
- [ ] Edge

### Screen Readers

- [ ] None
- [ ] NVDA
- [ ] JAWS
- [ ] VoiceOver on MacOS
- [ ] VoiceOver on iOS

**Comment by yuriy-fix** [comment]
@rolfsmeds , should selection be toggled in row-focus mode?

**Comment by web-padawan** [comment]
> should selection be toggled in row-focus mode?

When the selection column is the first one or is frozen (example in https://github.com/vaadin/flow-components/issues/6793), it already appears to work this way.
I think it's because switching to row focus mode requires navigating to first column and pressing Arrow Left.

**Comment by TatuLund** [comment]
Related to: https://github.com/vaadin/web-components/issues/5494

**Comment by rolfsmeds** [comment]
> [@rolfsmeds](https://github.com/rolfsmeds) , should selection be toggled in row-focus mode?

Yes, I do think it should.

**Comment by yuriy-fix** [comment]
The other question is how it should behave with tree grid. Should it toggle or expand an item in it? (Same question applicable to the item-details)
Based on the discussion, the expansion should happen instead of toggling.
